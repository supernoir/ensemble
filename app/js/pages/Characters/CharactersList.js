import React from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';

export default class CharactersList extends React.Component {
	constructor(){
		super();
		this.state = {
			characters: []
		};
	}
	componentDidMount(){
		axios.get('http://localhost:3030/characters')
			.then(res => {
				this.setState({ characters: res.data });
			}
			);
	}

	render(){
		return(
			<div>
				<ol class="breadcrumb">
					<li><Link to="/">Home</Link></li>
					<li class="active"><Link to="/characters">Characters</Link></li>
				</ol>

				<div class="jumbotron jumbo-characters">
					<h1>Characters</h1>
					<Link to="/addcharacter"><button type="submit" class="btn btn-default">Add Character</button></Link>
				</div>

				{this.state.characters.map(character => {
					return (
						<div class="panel panel-default">
							<div class="panel-heading">
								<Link to={`/character/${character._id}`}><h2>{character.first_name} <b>{character.last_name}</b></h2></Link>
								<h4>{character.gender} <span className="text-muted">{character.age}</span></h4>
							</div>
							<div class="panel-body">
								<ul>
									{character.series !== void 0
										? <li>
												  <b>Series:</b> <a href="/:cast">{character.series}</a>
											  </li>
										: null
									}
									{character.book !== void 0
										? <li>
												  <b>Book:</b> <a href="/:cast">{character.book}</a>
											  </li>
										: null
									}
									{character.family !== void 0
										? <li>
												  <b>Family:</b> <a href="/:cast">{character.family}</a>
											  </li>
										: null
									}
								</ul>
								<hr/>
								<a>{'Edit'}</a> {' | '} <a>{'Delete'}</a>
							</div>
						</div>
					);
				})}

			</div>
		);
	}
}