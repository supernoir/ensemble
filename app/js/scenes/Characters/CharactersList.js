import React from 'react';
import intl from 'react-intl-universal';
import { Link } from 'react-router-dom';
import { Container, Breadcrumb, Segment, Header, Card, Divider, Button } from 'semantic-ui-react';
import CharacterCard from '../../layout/CharacterCard';
import Loader from '../../basics/Loader';

export default class CharactersList extends React.Component {
	componentDidMount() {
		this.props.getCharacters();
	}

	getProjectById = id => {
		this.props.getProjectById(id);
	};

	render() {
		return this.props.loading
			? <Loader loading={this.props.loading} />
			:	<Container>
				<Breadcrumb>
					<Breadcrumb.Section link>
						<Link to="/">{intl.get('component.dashboard')}</Link>
					</Breadcrumb.Section>
					<Breadcrumb.Divider />
					<Breadcrumb.Section active>
						<Link to="/characters">{intl.get('entity.characters')}</Link>
					</Breadcrumb.Section>
				</Breadcrumb>

				<Segment>
					<Header as="h2">
						{intl.get('entity.characters')}
						<Header.Subheader>{intl.get('desc.characters')}</Header.Subheader>
					</Header>
					<Divider />
					<Button icon="add">
						<Link to="/addcharacter">{intl.get('character.action-add')}</Link>
					</Button>

				</Segment>

				<Divider />

				<Card.Group>
					{this.props.characters.map(character => {
						return (
							<CharacterCard
								character={character}
								projectTitle={this.props.project !== void 0 ? this.props.project.title : ''}
								getProjectTitle={() => this.getProjectById(character.project)}
							/>
						);
					})}
				</Card.Group>
			</Container>;
	}
}
