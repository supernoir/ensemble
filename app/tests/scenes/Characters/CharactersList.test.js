import React from 'react';
import { shallow } from 'enzyme';

// Import Mocks
import mockProject from '../../mocks/mockProject';
import mockCharacters from '../../mocks/mockCharacters';

// Import Components
import CharactersList from '../../../js/scenes/Characters/CharactersList';

test('CharactersList -- Snapshot Test', () => {
	const wrapper = shallow(
		<CharactersList
			loading={false}
			getCharacters={() => jest.fn()}
			characters={mockCharacters}
			getProjectById={id => jest.fn(id)}
			deleteSpecificCharacter={id => {
				jest.fn(id);
			}}
			project={mockProject}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});
