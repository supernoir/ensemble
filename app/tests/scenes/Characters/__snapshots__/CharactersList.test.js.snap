// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList -- Snapshot Test 1`] = `
<Container>
  <Breadcrumb>
    <BreadcrumbSection>
      <Link
        replace={false}
        to="/"
      />
    </BreadcrumbSection>
    <BreadcrumbDivider />
    <BreadcrumbSection
      active={true}
    >
      <Link
        replace={false}
        to="/characters"
      />
    </BreadcrumbSection>
  </Breadcrumb>
  <Segment>
    <Header
      as="h2"
    >
      <HeaderSubheader />
    </Header>
    <Divider />
    <Button
      as="button"
    >
      <Icon
        as="i"
        name="add"
      />
       
      <Link
        replace={false}
        to="/addcharacter"
      />
    </Button>
  </Segment>
  <Divider />
  <CardGroup>
    <CharacterCard
      character={
        Object {
          "_id": "1",
          "birthday": "2019-01-01",
          "desc": "Test1",
          "first_name": "Harry",
          "full_name": "Harry James Potter",
          "gender": "Male",
          "last_name": "Potter",
          "middle_name": "James",
          "origin": "Little Whinging, United Kingdom",
          "projects": Array [
            "1",
            "2",
          ],
        }
      }
      deleteSpecificCharacter={[Function]}
      getProjectTitle={[Function]}
      key="[object Object]-0"
    />
    <CharacterCard
      character={
        Object {
          "_id": "2",
          "birthday": "2015-02-02",
          "desc": "Test2",
          "first_name": "Hermione",
          "full_name": "Hermione Granger",
          "gender": "Female",
          "last_name": "Granger",
          "middle_name": "",
          "origin": "London, United Kingdom",
          "projects": Array [
            "2",
            "3",
          ],
        }
      }
      deleteSpecificCharacter={[Function]}
      getProjectTitle={[Function]}
      key="[object Object]-1"
    />
  </CardGroup>
</Container>
`;
